### PR TITLE
Consolidate separate functions for ions and electrons in plasma parameter calculations

### PR DESCRIPTION
**Pull request by RoberTnf**
Closing #106

The checklist of things to do are:

- [X]  Merge electron_thermal_speed and ion_thermal_speed into thermal_speed
- [x]  Merge electron_gyrofrequency and ion_gyrofrequency into gyrofrequency
- [x]  Merge electron_gyroradius and ion_gyroradius into gyroradius
- [x]  Merge electron_plasma_frequency and ion_plasma_frequency into plasma_frequency
- [x]  Merge electron_inertial_length and ion_inertial_length into inertial_length
- [x] add a parameter method to thermal_speed 
I will be updating as I progress with the tasklist.

### CODE REVIEW

**Comment by RoberTnf** [comment]
Why do I lack coverage in parts of the code that I didn't touch?

File: plasmapy/physics/parameters.py [hunk 1]
```diff
@@ -323,46 +323,46 @@ def ion_sound_speed(*ignore, T_e=0*units.K, T_i=0*units.K,
 @check_quantity({
     'T': {'units': units.K, 'can_be_negative': False}
 })
-def thermal_speed(T, ion="e"):
-    r"""Returns the most probable speed for an ion within a Maxwellian
-    distribution.
+def thermal_speed(T, particle="e"):
+    r"""Returns the most probable speed for an particle within a Maxwellian
+    distributparticle.
```
Comment (StanczakDominik) [inline plasmapy/physics/parameters.py]: What's a distributparticle? ;)

File: plasmapy/physics/parameters.py [hunk 2]
```diff
@@ -380,7 +380,7 @@ def thermal_speed(T, ion="e"):
     >>> from astropy import units as u
     >>> thermal_speed(5*u.eV, 'p')
```
Comment (StanczakDominik) [inline plasmapy/physics/parameters.py]: Do you think you could add an example for the default (electron) value here?

File: plasmapy/physics/parameters.py [hunk 3]
```diff
@@ -596,139 +596,65 @@ def gyroradius(B, *args, Vperp=None, T_i=None, ion='e'):
...
-
-
-@check_quantity({
-    'n_i': {'units': units.m**-3, 'can_be_negative': False}
+    'n': {'units': units.m**-3, 'can_be_negative': False}
 })
-def ion_plasma_frequency(n_i, ion='p'):
-    r"""Calculates the ion plasma frequency.
+def plasma_frequency(n, particle='e'):
+    r"""Calculates the particle plasma frequency.
```
Comment (StanczakDominik) [inline plasmapy/physics/parameters.py]: "Defaults to the fastest, electron plasma frequency.". How's that sound?

File: plasmapy/physics/parameters.py (hunk 1 above)
Comment (RoberTnf) [reply plasmapy/physics/parameters.py]: Is a distribution that also accepts electrons.

Comment (RoberTnf) [reply plasmapy/physics/parameters.py]: Is a distribution that also accepts electrons.

**Comment by RoberTnf** [comment]
I'm using VS Code, why?

**Comment by StanczakDominik** [comment]
Well, your renaming commits come really fast, it's nice to see this kind of proficiency :D

File: plasmapy/physics/parameters.py (hunk 1 above)
Comment (StanczakDominik) [reply plasmapy/physics/parameters.py]: Ooooh wait, I get it. It's like a distribution, but particle? Does this exhaust the pun limit for `physics`? :smile:

Comment (RoberTnf) [reply plasmapy/physics/parameters.py]: This is an accidental pun, it's what happens when you try to replace the mentions to `ion` with `particle`. I will fix it :smile:

File: plasmapy/physics/parameters.py (hunk 2 above)
Comment (RoberTnf) [reply plasmapy/physics/parameters.py]: I think that's is covered in line 386: `thermal_speed(5*u.eV)`

File: plasmapy/physics/parameters.py (hunk 3 above)
Comment (RoberTnf) [reply plasmapy/physics/parameters.py]: I don't understand what you mean.

File: plasmapy/physics/parameters.py (hunk 2 above)
Comment (StanczakDominik) [reply plasmapy/physics/parameters.py]: ... huh, it is. Sorry, I missed it somehow!

File: plasmapy/physics/parameters.py (hunk 3 above)
Comment (StanczakDominik) [reply plasmapy/physics/parameters.py]: I mean we could change the beginning of this docstring like this:

    """Calculates the particle plasma frequency. Defaults to the fastest, electron plasma frequency. 

The default behavior of this function seems important enough to mention it straight away :)

Comment (RoberTnf) [reply plasmapy/physics/parameters.py]: Ahh, Ok, sounds fine. Should something similar be done for every other function I modified?

**Comment by StanczakDominik** [comment]
I'm not sure whether it's really necessary for all of them. The plasma
frequency sort of stood out to me as I was just reading about it like two
days ago. I think we can live without changing it in most of these
functions!

**Comment by RoberTnf** [comment]
I think that's everything, unless I missed something!

**Comment by StanczakDominik** [comment]
Apparently we still have some conflicts on merge, probably because of the doctest thing I merged lately. Could you look into that? :)

**Comment by RoberTnf** [comment]
Ok, I'll try later!

**Comment by RoberTnf** [comment]
So after solving the conflicts I am not able to run pytest, it's trying to run docs/example.py, and obviously it can't. I suppose this is a separate issue but I'm not sure if it's something I did wrong.

**Comment by StanczakDominik** [comment]
Oh yeah, there was a fix in master to this one. Check out [these two lines](https://github.com/PlasmaPy/PlasmaPy/blob/master/.travis.yml#L21-L22) and make sure that's in your code.

**Comment by RoberTnf** [comment]
I closed this by accident, sorry, not used to github.

I have those lines, I've pulled master, but I get:

```
 $ pytest
============================= test session starts ==============================
platform linux -- Python 3.6.2, pytest-3.2.2, py-1.4.34, pluggy-0.4.0
rootdir: /home/roberto/git/PlasmaPy, inifile: pytest.ini
collected 706 items / 2 errors                                                  

==================================== ERRORS ====================================
__________________________ ERROR collecting setup.py ___________________________
/usr/lib/python3.6/distutils/core.py:134: in setup
    ok = dist.parse_command_line()
../../.virtualenvs/plasmapy/lib/python3.6/site-packages/setuptools/dist.py:442: in parse_command_line
    result = _Distribution.parse_command_line(self)
/usr/lib/python3.6/distutils/dist.py:490: in parse_command_line
    raise DistutilsArgError("no commands supplied")
E   distutils.errors.DistutilsArgError: no commands supplied

During handling of the above exception, another exception occurred:
setup.py:38: in <module>
    python_requires='>=3.6',
/usr/lib/python3.6/distutils/core.py:136: in setup
    raise SystemExit(gen_usage(dist.script_name) + "\nerror: %s" % msg)
E   SystemExit: usage: pytest [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
E      or: pytest --help [cmd1 cmd2 ...]
E      or: pytest --help-commands
E      or: pytest cmd --help
E   
E   error: no commands supplied
_______________________ ERROR collecting docs/example.py _______________________
docs/example.py:34: in <module>
    from my_module import my_func, other_func
E   ModuleNotFoundError: No module named 'my_module'
!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!
=========================== 2 error in 1.26 seconds ============================
```

**Comment by StanczakDominik** [comment]
Happens to everyone :) 

This is something we'll have to fix - as a workaround, run it as `pytest plasmapy` from the root of the repository for now. It should then get the doctest testing config from `pytest.ini` and avoid `docs`.

**Comment by StanczakDominik** [comment]
Aaaand congratulations @RoberTnf ! :D

**Pull request merged**